Marlin/src/pins/stm32f4/pins_CREALITY_CR4NTXXC10.h

```diff
@@ -0,0 +1,261 @@
+  #define X_SERIAL_RX_PIN                   PC7
+
+  #define Y_SERIAL_TX_PIN        X_SERIAL_TX_PIN
+  #define Y_SERIAL_RX_PIN        X_SERIAL_RX_PIN
+
+  #define Z_SERIAL_TX_PIN        X_SERIAL_TX_PIN
+  #define Z_SERIAL_RX_PIN        X_SERIAL_RX_PIN
+
+  #define E0_SERIAL_TX_PIN       X_SERIAL_TX_PIN
+  #define E0_SERIAL_RX_PIN       X_SERIAL_RX_PIN
+
+  #define E1_SERIAL_TX_PIN       X_SERIAL_TX_PIN
+  #define E1_SERIAL_RX_PIN       X_SERIAL_RX_PIN
+
+  #define X_DIAG_PIN                        PC13
+  #define Y_DIAG_PIN                        PC14
+  #define Z_DIAG_PIN                        PC15
+  #define E0_DIAG_PIN                       PA15
+
+#endif // HAS_TMC_UART
+
+//
+// SD Card
+//
+#define SD_DETECT_PIN                       PC1
+#define SDCARD_CONNECTION                ONBOARD
+#define ONBOARD_SPI_DEVICE                     1
+//#define ONBOARD_SD_CS_PIN                 PA4   // SDSS
+#define SDIO_SUPPORT
+#define NO_SD_HOST_DRIVE                  // This board's SD is only seen by the printer
+
+#if ENABLED(CR10_STOCKDISPLAY)
+
+  #if ENABLED(RET6_12864_LCD)
+    // RET6 12864 LCD
+    #define LCD_PINS_RS                     PB12
+    #define LCD_PINS_ENABLE                 PB15
+    #define LCD_PINS_D4                     PB13
+
+    #define BTN_ENC                         PB2
+    #define BTN_EN1                         PA2
+    #define BTN_EN2                         PB14
+
+    #ifndef HAS_PIN_27_BOARD
+      #define BEEPER_PIN                    PC0
+    #endif
+
+  #elif ENABLED(VET6_12864_LCD)
+    // VET6 12864 LCD
+    #define LCD_PINS_RS                     PA4
+    //#define LCD_PINS_ENABLE               PA7
+    #define LCD_PINS_D4                     PA5
+
+    #define BTN_ENC                         PC5
+    #define BTN_EN1                         PB10
+    #define BTN_EN2                         PA6
+
+  #else
+    #error "Define RET6_12864_LCD or VET6_12864_LCD to select pins for CR10_STOCKDISPLAY with the CREALITY_CR4NTXXC10 controller."
+  #endif
+
+#elif ENABLED(DWIN_CREALITY_LCD)
+
+  // RET6 DWIN ENCODER LCD
+  #define BTN_ENC                           PB14
+  #define BTN_EN1                           PB15
+  #define BTN_EN2                           PB12
+
+  //#define LCD_LED_PIN                     PB2
+  #ifndef BEEPER_PIN
+    #define BEEPER_PIN                      PB13
+    #undef SPEAKER
+  #endif
+
+#elif ENABLED(DWIN_VET6_CREALITY_LCD)
+
+  // VET6 DWIN ENCODER LCD
+  #define BTN_ENC                           PA6
+  //#define BTN_EN1                         PA7
+  #define BTN_EN2                           PA4
+
+  #define BEEPER_PIN                        PA5
+
+#endif
+
+//
+// M3/M4/M5 - Spindle/Laser Control
+//
+#if HAS_CUTTER
+  //#undef HEATER_0_PIN
+  //#undef HEATER_BED_PIN
+  //#undef FAN0_PIN
+  #define SPINDLE_LASER_ENA_PIN             PC0   // FET 1
+  #define SPINDLE_LASER_PWM_PIN             PC0   // Bed FET
+  #define SPINDLE_DIR_PIN                   PC0   // FET 4
```

Comment:
These don't seem quite correct. The ENA and PWM pins might coincide, but the DIR pin can't also be the same as the rest. Follow the pattern established by other (similar) pins files and enable SPINDLE/LASER pins based on whether SPINDLE or LASER is enabled. The file pins_CREALITY_V24S1_301.h is probably also incorrect.